identity/src/main/java/com/stripe/android/identity/IdentityActivity.kt

```diff
@@ -173,56 +165,49 @@ internal class IdentityActivity :
             }
         )
 
         identityViewModel.screenTracker.screenTransitionStart(
             startedAt = starterArgs.presentTime.asEpochMillisecondsClockMark()
         )
-
-        if (savedInstanceState == null || !savedInstanceState.getBoolean(
-                KEY_LAUNCHED_FALLBACK_URL,
-                false
-            )
-        ) {
-            // The Activity is newly created, set up navigation flow normally
-            setContentView(binding.root)
-            setUpNavigationController()
-        } else {
-            // The Activity is being recreated after being destroyed by OS.
-            // This happens when a fallback URL Activity is in front and IdentityActivity is destroyed.
-            // In this case, remove the NavHostFragment set earlier and let fallbackUrlLauncher return
-            // the callback to client.
-
-            // Recovered activity should already set up supportFragmentManager with a single NavHostFragment
-            require(supportFragmentManager.fragments.size == 1) {
-                "supportFragmentManager contains more than one fragment"
-            }
-            supportFragmentManager.beginTransaction().remove(supportFragmentManager.fragments[0])
-                .commit()
-        }
-    }
-
-    private fun setUpNavigationController() {
-        // hide supportActionBar and use the customized ToolBar to configure NavController.
-        // supportActionBar is unreliable as it might be null if host app uses a NoActionBar theme.
         supportActionBar?.hide()
 
-        navController =
-            (supportFragmentManager.findFragmentById(R.id.identity_nav_host) as NavHostFragment).navController
-
-        navController.setGraph(R.navigation.identity_nav_graph)
-
-        onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
+        setContent {
+            var topBarState by remember {
+                mutableStateOf(IdentityTopBarState.DEFAULT)
+            }
+            MdcTheme {
+                IdentityNavGraph(
+                    identityViewModel = identityViewModel,
+                    fallbackUrlLauncher = this,
+                    appSettingsOpenable = this,
+                    cameraPermissionEnsureable = this,
+                    verificationFlowFinishable = this,
+                    identityScanViewModelFactory = subcomponent.identityScanViewModelFactory,
+                    topBarState = topBarState,
+                    onTopBarNavigationClick = {
+                        onBackPressedCallback.handleOnBackPressed()
+                    },
+                ) {
+                    this.navController = it
+                    onBackPressedCallback =
+                        IdentityActivityOnBackPressedCallback(
+                            this,
+                            navController,
+                            identityViewModel
+                        )
+                    onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
```

Comment:
should this be under a `LaunchedEffect`? so that it doesn't get retriggered on recompositions.

Reply:
it is wrapped within a `LaunchedEffect` in `IdentityNavGraph`: [link](https://github.com/stripe/stripe-android/pull/5981/files#diff-8d1c09cc0e4ebccdb60d8dc29edb9b83e2bea76212c02687c05ca1044adbf479)